integrations/oci/metrics/metrics/src/main/java/io/helidon/integrations/oci/metrics/OciMetricsSupport.java

```diff
@@ -0,0 +1,527 @@
+            result.append("_").append(metricType.toString().replace(" ", "_"));
+
+            String units = formattedBaseUnits(metadata.getUnit());
+            if (units != null && !units.isBlank()) {
+                result.append("_").append(units);
+            }
+            return result.toString();
+        }
+    }
+
+    static String formattedBaseUnits(String metricUnits) {
+        String baseUnits = baseMetricUnits(metricUnits);
+        return baseUnits == null ? "" : baseUnits;
+    }
+
+    static String baseMetricUnits(String metricUnits) {
+        if (!MetricUnits.NONE.equals(metricUnits) && !metricUnits.isEmpty()) {
+            for (UnitConverter converter : UNIT_CONVERTERS) {
+                if (converter.handles(metricUnits)) {
+                    return converter.baseUnits();
+                }
+            }
+        }
+        return null;
+    }
+
+    private void shutdown(WebServer webServer) {
+        if (webServerCounter.decrementAndGet() < 1) {
+            LOGGER.fine("Shutting down OCI Metrics agent");
+            scheduledExecutorService.shutdownNow();
+        }
+    }
+
+    private void startExecutor() {
+        scheduledExecutorService = Executors.newScheduledThreadPool(1);
+        scheduledExecutorService.scheduleAtFixedRate(this::pushMetrics, initialDelay, delay, schedulingTimeUnit);
+    }
+
+    private void pushMetrics() {
+        List<MetricDataDetails> allMetricDataDetails = ociMetricsData.getMetricDataDetails();
+        LOGGER.finest(String.format("Processing %d metrics", allMetricDataDetails.size()));
+
+        if (allMetricDataDetails.size() > 0) {
+            while (true) {
+                if (allMetricDataDetails.size() > batchSize) {
+                    postBatch(allMetricDataDetails.subList(0, batchSize));
+                    // discard metrics that had been posted
+                    allMetricDataDetails.subList(0, batchSize).clear();
+                    if (batchDelay > 0L) {
+                        try {
+                            schedulingTimeUnit.sleep(batchDelay);
+                        } catch (InterruptedException ignore) {
+                        }
+                    }
+                } else {
+                    postBatch(allMetricDataDetails);
+                    break;
+                }
+            }
+        }
+    }
+
+    private void postBatch(List<MetricDataDetails> metricDataDetailsList) {
+        PostMetricDataDetails postMetricDataDetails = PostMetricDataDetails.builder()
+                .metricData(metricDataDetailsList)
+                .build();
+
+        PostMetricDataRequest postMetricDataRequest = PostMetricDataRequest.builder()
+                .postMetricDataDetails(postMetricDataDetails)
+                .build();
+
+        LOGGER.finest(String.format("Pushing %d metrics to OCI", metricDataDetailsList.size()));
+        if (LOGGER.isLoggable(Level.FINEST)) {
+            metricDataDetailsList
+                    .stream()
```

Comment:
`List` has its own `forEach`. No stream is needed here.

Reply:
Implementing